assume/reinforcement_learning/learning_role.py

```diff
@@ -182,14 +163,16 @@ def handle_message(self, content: dict, meta: dict) -> None:
             meta (dict): The metadata associated with the message. (not needed yet)
         """
 
-        if content.get("type") == "replay_buffer":
+        if content.get("type") == "save_buffer_and_update":
             data = content["data"]
             self.buffer.add(
                 obs=data[0],
                 actions=data[1],
                 reward=data[2],
             )
 
+        self.update_policy()
```

Comment:
The way you have sheduled the update policy now, right after the save to buffer, will also update the policy druing the inital exploration. This should not be a problem besides slowing the simualtion down right?

Reply:
the update_policy function anyway checks if we are still in the initial exploration, so no updates are actually executed

Reply:
Ah okay, it is weird that this is made sure of in two different ways, but then fine.